src/datasource.ts

```diff
@@ -68,113 +59,140 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
       timeField;
 
     const finalQuery = getTemplateSrv().replace(wholeQuery, options.scopedVars, 'csv');
-
-    // The Zui app is able to show its "Shapes:" count withut a separate query.
-    // Once we move the plugin to the Zealot client we should be able to do
-    // the same.
-    const shapeQuery = finalQuery + ' | by typeof(this) | count()';
-    const shapeCountObservable = await getBackendSrv().fetch<Array<{}>>({
-      method: 'POST',
-      url: this.url + '/query',
-      data: { query: shapeQuery },
-    });
-    const shapeCount = await lastValueFrom(shapeCountObservable);
-    if (shapeCount.data.length === 0) {
-      throw new Error('No data points found to plot in this time range');
-    } else if (shapeCount.data[0] > 1) {
+    const resultStream = await zedClient.query(finalQuery);
+    await resultStream.promise;
+
+    // Before we attempt to build a dataframe out of the query response to hand to
+    // Grafana for plotting, we'll check the shapes that were returned alongside the
+    // query response.
+    const shapes = resultStream.shapes as TypeRecord[];
+
+    // We'll reject any data that came back with multiple shapes since this makes
+    // it difficult/impossible to reliably copy the data points from the response
+    // into columns in Grafana's data frame.
+    if (shapes.length > 1 ) {
       throw new Error('More than one shape detected (consider using "cut" or "fuse")');
+
+    // If there were no shapes at all, it means there's no data to plot. This could
+    // have been caused by one of a few things.
+    //
+    // 1. The query/pool are valid, but the user just has the time range pickers in
+    //    Grafana set to a range where there's no data. This is innocuous.
+    //
+    // 2. The Time Field is not usable for some reason. A common example of this would
+    //    be if a user imported JSON data where the Time Field actually contains
+    //    string-typed values that look like timestamps, but since they're not of
+    //    the genuine Zed "time" type, the time-based filtering done in the query
+    //    using the values from Grafana's time range picker will fail to locate any
+    //    points. In the future we may come up with some way for the plugin to
+    //    transparently cope in such a situation (grafana-zed-datasource/issues/10)
+    //    but for now we'll just inform the user of what the plugin saw so they can
+    //    shape their data and try again.
+    //
+    // In order to tell the difference between these two cases we need to execute
+    // the following special query focused on just the Time Field.
+    } else if (shapes.length === 0) {
+      const timeCheckQuery = 'from "' +
+        pool +
+        '" | union(typeof(' +
+        timeField +
+        '))';
```

Comment:
I think a template literal would make this a little more readable.
```suggestion
      const timeCheckQuery = `from "${pool}" | union(typeof(${timeField}))`;
```